### PR TITLE
Rush: if a file outside of a project changes, build should rebuild all projects

### PR DESCRIPTION
**Pull request by nickpape-msft**
VSO 357588

### CODE REVIEW

**Comment by nickpape-msft** [comment]
@pgonzal @iclanton @qz2017 ping

File: apps/rush/src/utilities/PackageChangeAnalyzer.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import {
+  getPackageDeps,
+  IPackageDeps
+} from '@microsoft/package-deps-hash';
+
+import {
+  RushConfiguration
+} from '@microsoft/rush-lib';
+
+export const PACKAGE_DEPS_FILENAME: string = 'package-deps.json';
```
Comment (qz2017) [inline apps/rush/src/utilities/PackageChangeAnalyzer.ts]: >PACKAGE_DEPS_FILENAME [](start = 13, length = 21)

Use RushConstants.

File: apps/rush/src/utilities/PackageChangeAnalyzer.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      }
+    }
+    return undefined;
+  }
+
+  private _fileExistsInFolder(filePath: string, folderPath: string): boolean {
+    const relativePath: string = path.relative(folderPath, filePath);
+
+    // if the file exists in the folder, relativePath will not start with ".."
+    return relativePath.split(path.sep)[0] !== '..';
```
Comment (qz2017) [inline apps/rush/src/utilities/PackageChangeAnalyzer.ts]: >return relativePath.split(path.sep)[0] !== '..'; [](start = 4, length = 48)

This code does not work if on different drive

Comment (nickpape-msft) [inline apps/rush/src/utilities/PackageChangeAnalyzer.ts]: when would that case ever happen?

Comment (qz2017) [inline apps/rush/src/utilities/PackageChangeAnalyzer.ts]: It probably is not a problem here, if both input paths are relative.

---
In reply to: [116859507](https://github.com/Microsoft/web-build-tools/pull/194#discussion_r116859507) [](ancestors = 116859507)

**Comment by qz2017** [comment]
:clock1:

File: apps/rush/src/utilities/PackageChangeAnalyzer.ts (hunk 1 above)
Comment (nickpape-msft) [inline apps/rush/src/utilities/PackageChangeAnalyzer.ts]: Done

**Pull request merged**